### PR TITLE
feat: add message to across

### PR DESCRIPTION
**Pull request by mrice32**

### CODE REVIEW

File: contracts/SpokePool.sol [hunk 1]
```diff
@@ -16,6 +16,15 @@ import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 import "@openzeppelin/contracts/utils/math/SignedMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+interface AcrossMessageHandler {
```
Comment (nicholaspai) [inline contracts/SpokePool.sol:21]: nit add comment here about whats expected of integrators before mergimg

File: contracts/SpokePool.sol [hunk 2]
```diff
@@ -801,23 +867,25 @@ abstract contract SpokePool is
             // EIP-712 compliant hash struct: https://eips.ethereum.org/EIPS/eip-712#definition-of-hashstruct
             keccak256(
                 abi.encode(
-                    keccak256("UpdateRelayerFeeMessage(int64 newRelayerFeePct,uint32 depositId,uint256 originChainId)"),
-                    newRelayerFeePct,
+                    UPDATE_DEPOSIT_DETAILS_HASH,
```
Comment (nicholaspai) [inline contracts/SpokePool.sol:870]: +1

File: contracts/SpokePool.sol [hunk 3]
```diff
@@ -833,6 +901,22 @@ abstract contract SpokePool is
         require(isValid, "invalid signature");
     }
 
+    function _verifySlowFill(
```
Comment (nicholaspai) [inline contracts/SpokePool.sol:904]: Moving it here just for readability?

File: contracts/SpokePool.sol [hunk 4]
```diff
@@ -948,20 +1025,28 @@ abstract contract SpokePool is
...
                 IERC20Upgradeable(relayData.destinationToken).safeTransferFrom(
                     msg.sender,
                     relayData.recipient,
                     amountToSend
                 );
             else IERC20Upgradeable(relayData.destinationToken).safeTransfer(relayData.recipient, amountToSend);
         }
+
+        if (Address.isContract(relayData.recipient) && relayData.message.length > 0) {
+            AcrossMessageHandler(relayData.recipient).handleAcrossMessage(
```
Comment (nicholaspai) [inline contracts/SpokePool.sol:1044]: So if this fails, then relay is not filled at all and depositor can update the recipient/data, right?

Comment (mrice32) [reply contracts/SpokePool.sol:1044]: Exactly!

File: contracts/SpokePool.sol (hunk 3 above)
Comment (mrice32) [reply contracts/SpokePool.sol:904]: Yep. Just to make the flow super simple and similar in each of the fill methods -- check some single-line requirements and then process the fill.

**Pull request merged**